src/react-queries/queries/accountManagement.ts

```diff
@@ -2,67 +2,75 @@ import { createQueryKeys } from '@lukemorales/query-key-factory'
 import type { AccountId } from '@shapeshiftoss/caip'
 import { type ChainId, fromAccountId } from '@shapeshiftoss/caip'
 import type { HDWallet } from '@shapeshiftoss/hdwallet-core'
+import type { AccountMetadata } from '@shapeshiftoss/types'
 import { deriveAccountIdsAndMetadata } from 'lib/account/account'
 import { assertGetChainAdapter } from 'lib/utils'
 import { checkAccountHasActivity } from 'state/slices/portfolioSlice/utils'
 
-const getAccountIdsWithActivity = async (
+const getAccountIdsWithActivityAndMetadata = async (
   accountNumber: number,
   chainId: ChainId,
   wallet: HDWallet,
 ) => {
   const input = { accountNumber, chainIds: [chainId], wallet }
   const accountIdsAndMetadata = await deriveAccountIdsAndMetadata(input)
-  const [accountId] = Object.keys(accountIdsAndMetadata)
+  const [[accountId, accountMetadata]] = Object.entries(accountIdsAndMetadata)
 
   const { account: pubkey } = fromAccountId(accountId)
   const adapter = assertGetChainAdapter(chainId)
   const account = await adapter.getAccount(pubkey)
   const hasActivity = checkAccountHasActivity(account)
 
-  return { accountId, hasActivity }
+  return { accountId, accountMetadata, hasActivity }
 }
 
 export const accountManagement = createQueryKeys('accountManagement', {
-  accountIdWithActivity: (accountNumber: number, chainId: ChainId, wallet: HDWallet) => ({
-    queryKey: ['accountIdWithActivity', chainId, wallet.getDeviceID(), accountNumber],
+  accountIdWithActivityAndMetadata: (
+    accountNumber: number,
+    chainId: ChainId,
+    wallet: HDWallet,
+    walletDeviceId: string,
```

Comment:
Note things could get quite borked with Ledger caching-wise since its device ID is always the same and we could get into some sneaky states (e.g account number already detected as added but was added for a previous device, meaning it can't be added anymore).

Not related to this PR though but just keeping this here for reference 🙏🏽

Reply:
Absolute 🧠 I'll raise a ticket to address this specifically

Reply:
https://github.com/shapeshift/web/issues/6814